keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/DebianArtifactPublisher.kt

```diff
@@ -0,0 +1,91 @@
+package com.netflix.spinnaker.keel.artifact
+
+import com.netflix.frigga.ami.AppVersion
+import com.netflix.spinnaker.igor.ArtifactService
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactStatus
+import com.netflix.spinnaker.keel.api.artifacts.BuildMetadata
+import com.netflix.spinnaker.keel.api.artifacts.DebianSemVerVersioningStrategy
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.artifacts.GitMetadata
+import com.netflix.spinnaker.keel.api.artifacts.SpinnakerArtifact
+import com.netflix.spinnaker.keel.api.artifacts.VersioningStrategy
+import com.netflix.spinnaker.keel.api.plugins.ArtifactPublisher
+import com.netflix.spinnaker.keel.api.plugins.SupportedArtifact
+import com.netflix.spinnaker.keel.api.plugins.SupportedVersioningStrategy
+import com.netflix.spinnaker.keel.api.support.EventPublisher
+import com.netflix.spinnaker.keel.core.comparator
+import com.netflix.spinnaker.kork.exceptions.IntegrationException
+import org.springframework.stereotype.Component
+
+/**
+ * Built-in keel implementation of [ArtifactPublisher] that does not itself receive/retrieve artifact information
+ * but is used by keel's `POST /artifacts/events` API to notify the core of new Debian artifacts.
+ */
+@Component
+class DebianArtifactPublisher(
+  override val eventPublisher: EventPublisher,
+  private val artifactService: ArtifactService
+) : ArtifactPublisher<DebianArtifact> {
+  override val supportedArtifact = SupportedArtifact("deb", DebianArtifact::class.java)
+
+  override val supportedVersioningStrategies: List<SupportedVersioningStrategy<*>>
+    get() = listOf(
+      SupportedVersioningStrategy("deb", DebianSemVerVersioningStrategy::class.java)
+    )
+
+  override suspend fun getLatestArtifact(artifact: DeliveryArtifact): SpinnakerArtifact? =
+    artifactService
+      .getVersions(artifact.name)
+      .sortedWith(artifact.versioningStrategy.comparator)
+      .firstOrNull() // versioning strategies return descending by default... ¯\_(ツ)_/¯
+      ?.let { version ->
+        val normalizedVersion = if (version.startsWith("${artifact.name}-")) {
+          version.removePrefix("${artifact.name}-")
+        } else {
+          version
+        }
+        artifactService.getArtifact(artifact.name, normalizedVersion)
+      }
+
+  override fun getFullVersionString(artifact: SpinnakerArtifact): String =
+    "${artifact.name}-${artifact.version}"
```

Comment:
I'm not sure why, but for Debian we prefix the version with the package name.

Reply:
There's no real reason. It's a leftover inconsistency from when we only had debian packages and didn't think about saving the whole thing vs extracting just the version.